lib/Dancer2.pm

```diff
@@ -121,82 +121,1018 @@ __END__
+endpoints, allowing you to create dynamic paths.
+
+=head4 Named placeholders
+
+Named placeholders are the most common way of specifying variables
+in your path. You can then retrieve those variables with the
+C<route_parameters> keyword.
+
+    get '/:greeting' => sub {
+        my $greeting = route_parameters->get('greeting');
+    };
+
+If you go to the path F</hello>, the variable C<$greeting> will be
+B<hello>.
+
+You can have as many variables as you want:
+
+    get '/:greeting/:name' => sub {
+        my $greeting = route_parameters->get('greeting');
+        my $name     = route_parameters->get('name');
+    };
+
+If you use the same placeholder string, C<get> will only return the
+last match in the path.
+
+Since C<route_parameters> returns L<Hash::MultiValue>, you can use
+it as a hash directly as well:
+
+    get '/:greeting/:name' => sub {
+        my $greeting = route_parameters->{'greeting'};
+        my $name     = route_parameters->{'name'};
+
+        # or a hash slice
+        my ( $greeting, $name ) = @{route_parameters()}{qw<greeting name>};
+    };
+
+=head4 Splat (Anonymous Placeholders)
 
-=head2 Documentation Index
+The keyword C<splat> returns a B<list> of all the anonymous variables
+matched.
 
-Documentation on Dancer2 is split into several sections. Below is a
-complete outline on where to go for help.
+    get '/*' => sub {
+        my ($greeting) = splat;
+    };
+
+    get '/*/*' => sub {
+        my ( $greeting, $name ) = splat;
+    };
+
+You can use as many as you like, but remember it will B<always>
+return a list, never a scalar.
+
+=head4 Megasplat (Greedy Placeholders)
+
+The Megasplat is used as a way to capture as many matches as possible,
+otherwise known as I<greediness>. This way, Dancer will grab every
+path segment (delimited by forward slashes) separately.
+
+In this case, C<splat> will still return a list, but where a megasplat
+indicator is present (C<**>), it will provide an array reference.
+
+    get '/**' => sub {
+        my @segments = splat;
+        my ( $greeting, $name ) = @{ $segments[0] };
+    };
+
+The reason is simple: a regular match and a greedy match are not the
+same and they can be mixed together:
+
+    get '/*/**' => sub {
+        my ( $greeting, $parts ) = splat;
+        my ( $title, $name ) = @{$parts};
+    };
+
+You can, of course, mix all of these together.
+
+    get '/*/**/*/:name' => sub {...};
+
+=head4 Regular expressions
+
+Since this is Perl, we allow full power to those in need. You can use
+regular expression objects (created with C<qr>) to specify your route:
+
+    get qr{^/view/$} => sub {...};
+
+You will need to anchor your regular expressions yourself, in order to
+allow you to provide path segments in regular expressions too.
+
+    get qr{/view/} => sub {...};
+
+This will match both F</view/> and F</new/view/here>.
+
+Any capturing done in regular expressionos are available using the
```

Comment:
`s/expressionos/expressions/;`

Reply:
I can't find this in the most recent version.